ckanext/unaids/dataset_transfer/logic.py

```diff
@@ -0,0 +1,54 @@
+from datetime import datetime
+import logging
+
+from ckan import model, logic, authz
+import ckan.plugins.toolkit as toolkit
+from ckanext.unaids.dataset_transfer.model import (
+    DatasetTransferRequest,
+    STATUS_EMAILED,
+    STATUS_EMAIL_FAILED
+)
+
+log = logging.getLogger(__name__)
+
+
+def send_dataset_transfer_emails(dataset_id, recipient_org_id):
```

Comment:
This method is too long and too complex. You should extract some smaller pieces of logic into separate methods.
That's going to improve the readability and more importantly you'll be able to create unit tests for those smaller part of logic easily.